### PR TITLE
Feature/user apis

### PR DESCRIPTION
**Pull request by DrishikaPatil01**
Credential Apis and user apis
Session and User Specific Signing key changes

### CODE REVIEW

File: api/password-manager-api/handlers/forgotPassword.go [hunk 1]
```diff
@@ -1 +1,46 @@
...
+import (
+	"net/http"
+	"password-manager-service/database"
+	"password-manager-service/types"
+	"password-manager-service/utils"
+
+	"github.com/gin-gonic/gin"
+)
+
+func ForgotPassword(conn *database.DatabaseConnection) gin.HandlerFunc {
```
Comment (HarshitCd) [inline api/password-manager-api/handlers/forgotPassword.go:13]: What if I give some other person's email id what try to change the password will that be possible? If so isn't that a bad thing?

File: api/password-manager-api/database/sessionQueries.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+func (conn *DatabaseConnection) CreateSession(userId string) (sessionId string) {
```
Comment (HarshitCd) [inline api/password-manager-api/database/sessionQueries.go:10]: what does `(sessionId string)` do?

File: api/password-manager-api/handlers/credentials.go [hunk 3]
```diff
@@ -1 +1,206 @@
...
+	fn := func(c *gin.Context) {
+		var credentials types.CredentialData
+
+		if err := c.BindJSON(&credentials); err != nil {
+			c.JSON(http.StatusBadRequest, "could not process request")
+			return
+		}
+
+		//Validate Session
+		sessionToken := c.GetHeader("SessionToken")
```
Comment (HarshitCd) [inline api/password-manager-api/handlers/credentials.go]: extract `"SessionToken"` and store it in a variable.

File: api/password-manager-api/handlers/credentials.go [hunk 4]
```diff
@@ -1 +1,206 @@
...
+		if err = conn.AddCredential(userId, &credentials); err != nil {
+			fmt.Println("Error while adding credentials", err)
+			c.JSON(http.StatusInternalServerError, "Error while adding credentials")
+			return
+		}
+
+		//Update Session
+		conn.UpdateSession(sessionToken)
+
+		c.Writer.Header().Set("SessionToken", sessionToken)
```
Comment (HarshitCd) [inline api/password-manager-api/handlers/credentials.go]: Why are you sending the session token back?

File: api/password-manager-api/handlers/signup.go [hunk 5]
```diff
@@ -30,13 +32,20 @@ func Signup(conn *database.DatabaseConnection) gin.HandlerFunc {
...
+
+		user.Password = utils.EncryptUserPassword(user.Password)
 		if err := conn.AddUser(user); err != nil {
 			c.JSON(http.StatusInternalServerError, "error occured while adding user")
-		} else {
-			c.JSON(http.StatusOK, "added user")
 		}
 
+		signingKey := utils.GenerateSigningKey()
+		fmt.Println("Signing key : ", signingKey)
```
Comment (HarshitCd) [inline api/password-manager-api/handlers/signup.go]: Remove this print statement

File: api/password-manager-api/handlers/forgotPassword.go (hunk 1 above)
Comment (DrishikaPatil01) [reply api/password-manager-api/handlers/forgotPassword.go:13]: So u suggest we take both username and email in the request?

Comment (HarshitCd) [reply api/password-manager-api/handlers/forgotPassword.go:13]: Implement forgot password just like any other web application.

File: docker-compose.yml [hunk 6]
```diff
@@ -34,6 +34,11 @@ services:
       REDIS_LINK: cache:6379
       REDIS_PASSWORD: cache
       SECRET_KEY: N1PCdw3M2B1TfJhoaY2mL736p2vCUc47
+      EMAIL: dpprojects952@gmail.com
+      EMAIL_KEY: zfdg hlkl xggu udig
```
Comment (HarshitCd) [inline docker-compose.yml]: Is it wise to put these two fields on a repo? 😂

Comment (DrishikaPatil01) [reply docker-compose.yml]: I created a special dummy account for this, so why not

File: api/password-manager-api/database/sessionQueries.go (hunk 2 above)
Comment (DrishikaPatil01) [reply api/password-manager-api/database/sessionQueries.go:10]: we are creating a new session in this method so we need to return the sessionId or token so this is that

File: docker-compose.yml (hunk 6 above)
Comment (DrishikaPatil01) [reply docker-compose.yml]: Ill remove it since ur not using it for testing it out

File: api/password-manager-api/handlers/credentials.go (hunk 4 above)
Comment (DrishikaPatil01) [reply api/password-manager-api/handlers/credentials.go]: Will remove it

File: api/password-manager-api/handlers/credentials.go (hunk 3 above)
Comment (DrishikaPatil01) [reply api/password-manager-api/handlers/credentials.go]: Are u suggesting to extract it commonly for every handler or to keep this variable like a common variable for the entire module?

Comment (HarshitCd) [reply api/password-manager-api/handlers/credentials.go]: Yep yep

File: docker-compose.yml (hunk 6 above)
Comment (HarshitCd) [reply docker-compose.yml]: Well the repo is public and anyone can use it mate XD

**Review by HarshitCd — APPROVED** [review]

**Pull request merged**